templates/_macros/hero.jinja

```diff
@@ -0,0 +1,109 @@
+{% from "_macros/cta/cta-block.jinja" import cta_block %}
+
+# Params
+# title_text: Hero title text (required)
+# subtitle_text: Hero subtitle text
+# is_full_width_on_tablet: whether image is in its own row on tablet.
+#   If false, image will be in its own row (stacked below header/description)
+# image_src: image source URL (required)
+# image_alt: image alt text
+# image_aspect_ratio: aspect ratio identifier of the image, from the `p-image-container` component
+# image_full_cover: whether the image should span the full width of the bottom of the hero section
+# image_hidden_breakpoints: list of breakpoints where the image should be hidden, e.g. ["large", "medium"] to only show on small screens
+# show_cta: whether to show the call-to-action block
+# cta_primary_text: primary CTA button text, shown in a primary-colored button
+# cta_primary_url: primary CTA button URL
+# cta_secondary_text: secondary CTA button text, shown as a link
+# cta_secondary_url: secondary CTA button URL
+# Slots
+# description: slot for additional paragraph-style content below the title and subtitle
+
+{% macro hero(
+  title_text="",
+  subtitle_text="",
+  is_full_width_on_tablet = true,
+  image_src = "",
+  image_alt = "",
+  image_aspect_ratio = "",
+  image_full_cover = false,
+  image_hidden_breakpoints = [],
+  show_cta = false,
+  cta_primary_text = "Learn more",
+  cta_primary_url = "#",
+  cta_secondary_text = "",
+  cta_secondary_url = "#"
+)-%}
+  {% set has_subtitle = subtitle_text|trim|length > 0 %}
+  {% set description_content = caller('description') %}
+  {% set has_description = description_content|trim|length > 0 %}
+  {% set has_specific_image_aspect_ratio = image_aspect_ratio|trim|length > 0 %}
+  {% set has_cta = show_cta and cta_primary_text|trim|length > 0 %}
+
+  {% macro hero_title_block() -%}
+    <h1 class="{% if image_full_cover %}u-no-margin--bottom{% endif %}">
```

Comment:
As far as I'm aware there is no need for the title to contain arbitrary HTML. Though if we did need it to, we could  use `title_text | safe`?

This one is also a bit complicated as I need to set `u-no-margin-bottom` on the header in full-cover mode (to prevent the h1-h2 being so far from eachother as [Lyubo noticed](https://github.com/canonical/vanilla-framework/pull/5189#issuecomment-2233829522)); so I need control over the element tag itself rather than letting the user pass it in.

Reply:
Hmm… that's a tricky one.

We have to allow HTML in titles (in terms of inline styling `<code>` or things like that).

But, in case of hero at least, we don't need a flexibility of changing level of headings. They are always gonna be h1 (mandatory title), and h2-sized paragraph (subtitle). We don't have to (and we don't want to) give devs tools to change those, so they can just pass text. Which in 95% cases will be just a string.

So… kind of starts to make sense to make them params instead of slots?
Of course we can also make them slots for content **inside** h1/h2, but that will be inconsistent with other slots where you are providing the "wrapping" elements, like paragraphs etc.

Let's discuss it.